narwhals/_dask/utils.py

```diff
@@ -44,17 +44,42 @@ def parse_exprs_and_named_exprs(
     for expr in exprs:
         if hasattr(expr, "__narwhals_expr__"):
             _results = expr._call(df)
+            _names = expr._output_names
         elif isinstance(expr, str):
             _results = [df._native_dataframe.loc[:, expr]]
+            _names = [expr]
         else:  # pragma: no cover
             msg = f"Expected expression or column name, got: {expr}"
             raise TypeError(msg)
-        for _result in _results:
-            results[_result.name] = _result
+
+        results.update(dict(zip(_names, _results)))
+
     for name, value in named_exprs.items():
         _results = value._call(df)
         if len(_results) != 1:  # pragma: no cover
             msg = "Named expressions must return a single column"
             raise AssertionError(msg)
         results[name] = _results[0]
+
     return results
+
+
+def parse_series(named_series: dict[str, Any]) -> tuple[Any, dict[str, Any]]:
+    de = get_dask_expr()
+    series = list(named_series.values())
+    lengths = [
+        1 if isinstance(s, de._collection.Scalar) else len(s.index) for s in series
```

Comment:
I don't think we should be doing `len(s.index)` for Dask, else we break the lazy nature

Dask is a bit unusual in that although it's lazy, there are some commands that trigger compute. I'd like to avoid them here

could we just have:
- check if objects are Dask Scalars. If so, call `to_series` on all of them and concatenate them
- if they're all Dask Series, just concatenate them (other checks, and the reduced API, should ensure that they all have the same index)
- raise if there's a mix of scalars and series

Reply:
I will take some time and try to rethink it from the ground up. Mind keeping the PR open for the sake of visibility?